### DISCUSSION TITLE
Including commit messages from dependent libraries.

### DISCUSSION THREAD

**Discussion by CaffeinatedCodeMonkey** [jscutlery/semver#293] [Ideas]
**Note**: This is based on the assumption that https://github.com/jscutlery/semver/pull/278 has been merged and dependent libraries can now cause bumps in the targeted project.

Commit messages from dependent libraries are not included in the targeted project's change logs when `--use-deps` is used. This is because the change logs are generated based on commits affecting the targeted project's directory. Unfortunately, the `standard-version` library doesn't support a means of including commits from more than one directory.

**Implementation Idea**
In the event that including those messages is in demand, then one possible avenue of is to make new commits to some file (what file would have to be determined) in the targeted project. There should likely be a commit for each level of version imposed by the dependencies. Each of those commits could include the commit messages that correspond to the particular increment level (patch, minor, or major).

**Comment by edbzn** [comment]
This implies breaking down `standard-version` into smaller functions to have full control over the release process. This is a major change and needs design + RFC. It's definitely something we should think about.
